File: SmartApp.Core/Contracts/IRepositoryBase.fs
namespace SmartApp.Core.Contract

open System.Threading.Tasks



type IRepositoryBase<'T> =
    
    abstract member Get: id:int64-> 'T

    abstract member GetAsync: id:int64 -> Task<'T>

    abstract member GetAll: skip:int -> take:int -> seq<'T>

    abstract member GetAllAsync: skip:int -> take:int  -> Task<int * seq<'T>>


    

File: SmartMoneyApp.Common/Interfaces/Core/IRepositoryBase.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace SmartApp.Common.Interfaces.Core
{
  public interface IRepositoryBase<T>
    {
        public T Get(long id);
        public Task<T> GetAsync(long id);
        public IEnumerable<T> GetAll(int skip, int take);
        public Task<Tuple<int, IEnumerable<T>>> GetAllAsync(int skip, int take);
        public Task Insert(T item);
        public void Update(T item);
        public Task SaveChangesAsync();
        public void Delete(T entity);
        public void Delete(IEnumerable<T> entities);
    }
}